weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooksCallRefExpansion.ts

```diff
@@ -29,115 +29,36 @@ export const useClientSideCallRefExpansion = (
   const [isExpanding, setIsExpanding] = useState(false);
 
   useEffect(() => {
+    let mounted = true;
     if (calls.loading || calls.result == null) {
       setExpandedCalls([]);
       setIsExpanding(true);
       return;
     }
 
-    const doExpansionIteration = async (
-      traceCalls: traceServerClientTypes.TraceCallSchema[]
-    ) => {
-      const refsNeeded = new Set<string>();
-      const expandedRefColumnsList = Array.from(expandedRefColumns ?? []);
-      traceCalls.forEach(call => {
-        expandedRefColumnsList.forEach(col => {
-          const colParts = col.split('.');
-          let value: any = call;
-          for (const part of colParts) {
-            while (
-              typeof value === 'object' &&
-              value != null &&
-              EXPANDED_REF_VAL_KEY in value
-            ) {
-              value = value[EXPANDED_REF_VAL_KEY];
-            }
-            if (value == null) {
-              break;
-            }
-            if (typeof value !== 'object' || !(part in value)) {
-              value = null;
-              break;
-            }
-            value = value[part];
-          }
-          while (
-            typeof value === 'object' &&
-            value != null &&
-            EXPANDED_REF_VAL_KEY in value
-          ) {
-            value = value[EXPANDED_REF_VAL_KEY];
-          }
-          if (isExpandableRef(value)) {
-            refsNeeded.add(value);
-          }
-        });
-      });
-
-      const refsNeededArray = Array.from(refsNeeded);
-
-      if (refsNeededArray.length === 0) {
-        setExpandedCalls(traceCalls);
+    if (calls.result.length === 0) {
```

Comment:
All this PR does is extract `const doExpansionIteration = async (` into a top-level function and call it from the hook here. The only logical change is `if (calls.result === callResultStart && mounted) {` conditional on update